src/installer/installer.go

```diff
@@ -355,3 +366,56 @@ func (i *installer) cleanupInstallDevice() error {
 
 	return i.ops.RemovePV(i.Device)
 }
+
+func (i *installer) verifyHostCanMoveToConfigurationStatus(inventoryHostsMapWithIp *map[string][]string) {
+	logs, err := i.ops.GetMCSLogs()
+	if err != nil {
+		i.log.Infof("Failed to get MCS logs, will retry")
+		return
+	}
+	_inventoryHostsMapWithIp := *inventoryHostsMapWithIp
+	for hostId, ips := range _inventoryHostsMapWithIp {
+		pat := fmt.Sprintf("(%s)", strings.Join(ips, "|"))
+		pattern, err := regexp.Compile(pat)
+		if err != nil {
+			i.log.WithError(err).Errorf("Failed to compile regex from host %s ips list", hostId)
+			return
+		}
+
+		if pattern.MatchString(logs) {
+			i.log.Infof("Host %s found in mcs logs, moving it to %s state", hostId, Configuring)
+			if err := i.inventoryClient.UpdateHostStatus(Configuring, hostId); err != nil {
+				i.log.Errorf("Failed to update node installation status, %s", err)
+				continue
+			}
+			delete(*inventoryHostsMapWithIp, hostId)
+		}
+	}
+}
+
+// will run as go routine and tries to find nodes that pulled ignition from mcs
+// it will get mcs logs of static pod that runs on bootstrap and will search for matched ip
+// when match is found it will update inventory service with new host status
+func (i *installer) updateConfiguringStatus(done <-chan bool) {
+	i.log.Infof("Start waiting for configuring state")
+	ticker := time.NewTicker(generalWaitTimeout)
+	var inventoryHostsMapWithIp map[string][]string
+	var err error
+	for {
+		select {
+		case <-done:
+			i.log.Infof("Exiting updateConfiguringStatus go routine")
+			return
+		case <-ticker.C:
+			i.log.Infof("searching for hosts that had pulled ignition already")
+			if inventoryHostsMapWithIp == nil {
+				inventoryHostsMapWithIp, err = i.inventoryClient.GetEnabledIdsIps()
+				if err != nil {
+					i.log.Warnf("Failed to get hosts info from inventory, err %s", err)
+					continue
+				}
+			}
+			i.verifyHostCanMoveToConfigurationStatus(&inventoryHostsMapWithIp)
```

Comment:
meaning u gonna read this file every time? no -f option?
btw, what will happen for workers & bootstrap?

Reply:
First of all file is really small. Verified it. So didn't really tried to solved this issue.

Regarding workers and bootstrap.
All nodes that will finish reboot and pull ignition from mcs that run on bootstrap will be moved to configuring (if name is ok) state by assisted-installer.
For others we have another ticket, we will need to implement same stuff in assisted-controller, that will read logs of mcs pod in openshift cluster